sar_pipeline/pipelines/isce3_rtc/cli.py

```diff
@@ -884,3 +909,203 @@ def make_rtc_opera_stac_and_upload_bursts(
+)
+@click.option(
+    "--s3-bucket",
+    required=False,
+    default="deant-data-public-dev",
+    type=str,
+    help="S3 where outputs are being stored. Required if s3 folders are set as input",
+)
+@click.option(
+    "--out-folder",
+    required=False,
+    default=PROJECT_ROOT_PATH / "comparison",
+    type=click.Path(file_okay=False, path_type=Path),
+    help="Folder to write the outputs of the comparison to",
+)
+def compare_products(
+    product,
+    local_product_folder_1,
+    local_product_folder_2,
+    s3_product_folder_1,
+    s3_product_folder_2,
+    s3_bucket,
+    out_folder,
+):
+    logger.info(f"The product being compared is : {product}")
+    logger.info(f"The outputs will be written to : {out_folder}")
+
+    if not out_folder.exists():
+        raise ValueError(
+            f"Specified --out-folder does not exist, please create: {out_folder}"
+        )
+
+    if not (s3_product_folder_1 or local_product_folder_1):
+        logger.error(
+            "Ensure either --local-product-folder-1 or --s3-product-folder-1 set"
+        )
+        raise ValueError
+    elif not (s3_product_folder_2 or local_product_folder_2):
+        logger.error(
+            "Ensure either --local-product-folder-2 or --s3-product-folder-2 set"
+        )
+        raise ValueError
+
+    if s3_product_folder_1 or s3_product_folder_2:
+        # set up the AWS download util
+        S3Downloader = S3Util()
+
+    if s3_product_folder_1:
+        # download product folder to local folder
+        local_product_folder_1 = out_folder / "folder_1" / s3_product_folder_1
+        os.makedirs(local_product_folder_1, exist_ok=True)
+        S3Downloader.download_folder(
+            s3_bucket, s3_product_folder_1, local_product_folder_1
+        )
+    if s3_product_folder_2:
+        # download product folder to local folder
+        local_product_folder_2 = out_folder / "folder_2" / s3_product_folder_2
+        os.makedirs(local_product_folder_2, exist_ok=True)
+        S3Downloader.download_folder(
+            s3_bucket, s3_product_folder_2, local_product_folder_2
+        )
+
+    # compare count of files and filetypes in folder
+    files_are_different, file_differences = compare_product_folder_files(
+        local_product_folder_1, local_product_folder_2
+    )
+    file_differences_path = out_folder / "file_differences.json"
+    if files_are_different:
+        logging.warning(
+            f"Differences in product files found. Saving summary to : {file_differences_path}"
+        )
+    else:
+        logger.info(
+            f"No differences in product files identified. Saving summary to : {file_differences_path}"
+        )
+    write_diffs_to_json(file_differences, file_differences_path)
+
+    # compare the json files
+    json_files_1 = list(local_product_folder_1.glob("*.json"))
+    json_files_2 = list(local_product_folder_2.glob("*.json"))
+    if len(json_files_1) > 1 or len(json_files_2) > 1:
```

Comment:
In what case would there be multiple json files? And what does it mean to compare first? Can you be certain that it's fair to compare the first json in the list from each folder?